doc/hardware/peripherals/nvmem.rst

```diff
@@ -0,0 +1,419 @@
+.. _nvmem_api:
+
+Non-Volatile Memory (NVMEM)
+###########################
+
+The NVMEM device driver model provides a uniform way for NVMEM consumers
+to access non-volatile memory devices such as EEPROMs, OTP/eFuse, FRAM,
+and battery-backed RAM (NVMEM providers).
+
+The API is provider-centric: drivers expose basic read/write operations
+and report their total size. Providers are responsible for enforcing
+device semantics (e.g., read-only policy, hardware-defined one-time
+programming for OTP/eFuse). Optionally, providers can expose configuration
+descriptors for consumer and tool diagnostics.
+
+.. contents::
+    :local:
+    :depth: 2
+
+
+Goals and vision
+****************
+
+The NVMEM model unifies access to diverse non-volatile memories behind a tiny,
+consistent API so application and driver consumers can:
+
+- Treat NVMEM like a simple byte-addressable space (read/write/get_size).
+- Discover device semantics at runtime (type and read-only via nvmem_info).
+- Describe reusable named “cells” in devicetree and bind consumers to them.
+
+Providers retain full control of hardware semantics and safety (e.g., enforcing
+OTP bit programming rules, honoring write-protect pins, respecting page/word
+granularity). The long-term vision is a single, cross-silicon way to read
+calibration data, MAC addresses, keys, serial numbers, and platform state.
+
+
+Key concepts
+************
+
+- NVMEM provider:
+  A device driver that implements the NVMEM API for a specific storage (e.g.,
+  I2C EEPROM, on-chip eFuse/OTP, FRAM, battery-backed SRAM).
+
+- NVMEM consumer:
+  Any code that reads/writes NVMEM data. This includes applications,
+  subsystems (e.g., networking fetching a MAC address), and manufacturing tools.
+
+- NVMEM cell:
+  A named region within a provider’s address space described in devicetree using
+  a fixed layout. Cells allow consumers to bind to stable names instead of hardcoding
+  offsets and sizes.
+
+
+NVMEM Provider API
+******************
+
+Zephyr's NVMEM provider API is implemented by device drivers for specific
+non-volatile memories (e.g., AT24/AT25 EEPROMs, NXP OCOTP eFuse). Consumers
+interact with these providers via a small set of functions:
+
+- :c:func:`nvmem_read` – Read bytes from the provider address space.
+- :c:func:`nvmem_write` – Write bytes to the provider (if permitted).
+- :c:func:`nvmem_get_size` – Get the total size (in bytes).
+- :c:func:`nvmem_get_info` – Optionally obtain a provider descriptor.
+  (:c:struct:`nvmem_info`) including storage type and current read-only state.
```

Comment:
I'm sorry, but you're changing the core idea of the initial NVMEM subsystem. the ability to extend and use different driver APIs.

What about BBRAM/flash/EEPROM/..., these can all be NVMEM providers Are you going to convert them all the NVMEM?